docs/api.md

```diff
@@ -0,0 +1,551 @@
+`VantiqResponse`
+
+The count property of the VantiqResponse will provide the number of objects deleted.
+
+#### Examples
+
+```python
+vr: VantiqResponse = await client.delete('myType', {'name': 'some name'}
+if vr.is_success:
+    print('Count of items deleted: ', vr.count)
+```
+### Vantiq.delete\_one() (async)
+Delete a single item from a resource.
+
+#### Parameters
+
+* resource : str -- The name of the Vantiq resource from which to delete.
+* resource_id : str -- The key for the item to delete.
+
+#### Returns
+
+`VantiqResponse`
+
+
+### Vantiq.insert() (async)
+Insert an item into Vantiq Resource.
+
+#### Parameters
+
+* resource : str -- Name of the Vantiq resource into which to insert.
+* instance : dict -- The values to be inserted.
+
+#### Returns
+
+`VantiqResponse`
+
+
+### Vantiq.upsert() (async)
+
+Upsert an item into Vantiq Resource.
+
+#### Parameters
+
+* resource : str --  Name of the Vantiq resource into which to upsert.
+* instance : dict -- The values to be upserted.
+
+#### Returns
+
+`VantiqResponse`
+
+### Vantiq.update() (async) 
+
+Update an item in a  Vantiq Resource.
+
+#### Parameters
+
+* resource : str -- Name of the Vantiq resource in which to update.
+* resource_id : str -- The key of the record to look up for replacement.  The _id property can be used.
+* instance : dict -- The values to be updated.
+
+#### Returns
+
+`VantiqResponse`
+
+### Vantiq.download() (async)
+Download content from a Vantiq Document, Image, Video, or TensorflowModel.
+
+#### Parameters
+
+* path : str -- The path from which to obtain the data. This is found in the `content` field of the base object.
+
+#### Returns
+
+`VantiqResponse`. In this case, the response's `body` field is an `aiohttp.StreamReader` that can be used to fetch the content returned.
+
+#### Examples
+
+```python
+vr: VantiqResponse = await client.select_one(VantiqResources.DOCUMENTS, 'mydoc')
+mydoc = vr.body
+if vr.is_success:
+    download_vr: VantiqResponse = await client.download(mydoc.content)
+    if download_vr.is_success:
+        reader = download_vr.body
+        while True:
+            data = await reader.read(100)  # Read our data 100 bytes at a time
+            if len(data) == 0:
+                break
+                # Do something with the data
+                ...
+```
+
+### Vantiq.upload() (async) 
+Upload a file (or in-memory data), creating a object to hold that data.
+
+This allows the upload of a file to create a document, inage, video, or tensorflow model.
+
+#### Parameters
+   
+* resource : str -- The Vantiq resource type to create.
+* content_type : str -- The type of data contained in the file.  This will be set as the `contentType` of the resulting object.
+* filename : str -- The name of the file to upload.
+* doc_name : str -- (optional) The name of the object to create. If this is missing, use the filename.
+* inmem : str | bytes | bytearray -- (optional) Content to be uploaded.  Used with the content is not in a file.
```

Comment:
unclear -- "used with" -- should this be "used when"?  If this option is used do you still need to provide a value for `filename`?  Does that just become what `doc_name` is in the file upload case (the name of the target resource instance on the Vantiq side).

Reply:
yes.  That's right.  If you have in-memory data, you can used either `filename` or `doc_name` to name the document.  Providing both results in an error since there's not real "file" to read.  Updating the api.md text.